src/plugins/vis_type_vislib/public/vislib/lib/axis/axis_scale.js

```diff
@@ -38,10 +38,24 @@ export class AxisScale {
     });
   }
 
-  getTimeDomain(data) {
+  getDomainExtent(data) {
     return [this.minExtent(data), this.maxExtent(data)];
   }
 
+  getOrdinalDomain(data = []) {
+    if (this.ordered?.interval !== undefined) {
+      const [min, max] = this.getDomainExtent(data);
+
+      // This is an arbetrary limit to prevent arrays larger than 500 elements
+      // At which point, the interval would not be perceptible
+      if ((max - min) / this.ordered?.interval <= 500) {
+        return _.range(min, max + this.ordered?.interval, this.ordered?.interval);
+      }
+    }
```

Comment:
The alternative to setting this limit:

- Remove the fix and use the wrong interval, but generally only seen on split chart aggregations.
- Change the scale to use a custom linear band scale and not the current ordinal scale.
- Have elasticsearch return all buckets. Pretty much the same impact as this fix and more data over the network.

Setting this limit allows _most_ of the problematic visualizations to render the correct interval. Not doing this fix renders all problematic visualizations with the wrong interval.